src/AMSlib/wf/basedb.hpp

```diff
@@ -938,54 +928,28 @@ class EventBuffer
                 nbyte_drained);
       }
       evbuffer_unlock(buffer);
-
-      std::string result =
-          std::to_string(self->_rank) + ":";
-      for (int i = 0; i < k - 1; i++) {
-        result.append(std::to_string(data[i]) + ":");
-      }
-      result.append(std::to_string(data[k - 1]) + "\n");
-      
-      // For resiliency reasons we encode the result in base64
-      // Not that it increases the size (n) of messages by approx 4*(n/3)
-      std::string result_b64 = self->encode64(result);
-      DBG(RabbitMQDB,
-          "[rank=%d] #elements (float/double) = %d, stringify size = %d, size in base64 "
-          "= %d",
-          self->_rank,
-          k,
-          result.size(),
-          result_b64.size())
-      if (result_b64.size() % 4 != 0) {
-        WARNING(EventBuffer,
-                "[rank=%d] Frame size (%d elements)"
-                "cannot be %d more than a multiple of 4!",
-                self->_rank,
-                result_b64.size(),
-                result_b64.size() % 4)
-      }
-
+      double mbyte_sent = nbyte_drained / (1024.0 * 1024.0);
       // publish a message via the reliable-channel
-      self->_rchannel->publish("", self->_queue, result_b64)
-          .onAck([self, nbyte_drained]() {
+      self->_rchannel->publish("", self->_queue, reinterpret_cast<char*>(data.get()), datlen)
+          .onAck([self, mbyte_sent, nbyte_drained]() {
```

Comment:
Is it necessary to keep track of the number of bytes being sent? Doesn't RMQ provide guarantees of delivering an entire message?

Reply:
No it is necessary, that was mostly for debug purposes. I can simplify the code and remove that.

Reply:
s.g.